app/javascript/mastodon/components/status_action_bar.js

```diff
@@ -57,6 +58,13 @@ export default class StatusActionBar extends ImmutablePureComponent {
     this.props.onReply(this.props.status, this.context.router.history);
   }
 
+  handleShareClick = () => {
+    navigator.share({
+      text: this.props.status.get('search_index'),
```

Comment:
Wanted to avoid the stripping of HTML tags inside a click handler. Will update it.